keps/0032-community-repository-management.md

```diff
@@ -0,0 +1,108 @@
+
+### Goals
+
+- Provide a simple workflow for upstream operator packages to get added to the community repository
+- Remove the need to "host" copies of upstream operator packages in a Git repository
+- Provide a mechanism to define upstream operator package sources
+- Provide a mechanism to define upstream operator package maturity
+
+### Non-Goals
+
+- Change the practice of creating a PR against a Git repository to add operator packages to the community repository
+- Change operator package management as defined in [KEP-10](0010-package-manager.md) and [KEP-15](0015-repository-management.md)
+- Integration with [ArtifactHub](https://artifacthub.io/)
+
+## Proposal
+
+Upstream operator developers still create PRs against a Git repository to add their operator packages to the community repository. This Git repository lists references to upstream operator packages instead of a full copy of an operator package. A reference can point to a Git resitory or a package tarball. A version of an operator package is described by a specific tag of a Git repository or a URL pointing to an operator tarball of that release.
+
+For example, consider an operator package developed at `github.com/example/example-operator` that has  tagged versions `1.0.0`, `1.1.0` and the operator package in the `operator` folder. There's also a tarball for version `0.9.0` hosted at `example.org/example-operator-0.9.0.tgz`. To add or update this operator package, the developers would create a PR referencing their upstream Git repository and the specific version, e.g. by adding a file `example-operator.yaml` like
+
+```yaml
+api: v1alpha1
+name: Example Operator
+git-sources:
+- name: git-repo
+  url: github.com/example/example-operator.git
+versions:
+- version: "0.9.0"
+  url: example.org/example-operator-0.9.0.tgz
+- version: "1.0.0"
+  git:
+    source: git-repo
+    tag: "1.0.0"
+    dir: operator
+- version: "1.1.0"
+  git:
+    source: git-repo
+    tag: "1.1.0"
+    dir: operator
+```
+
+Once this PR is merged, CI tooling detects the new YAML file, clones the referenced upstream Git repository, checks out the tag, and adds the operator package in the specified folder to the existing index. This workflow is similar to [krew-index](https://github.com/kubernetes-sigs/krew-index). Of course, CI tests that don't update the community repository can run before the PR is merged. These tests include checking the referenced operator package for validity. Additional conformance testing can be added as well.
+
+We can add more metadata to the YAML reference file. E.g., support for different upstream sources like Mercurial.
+
+The maturity level of an operator package can be provided in this file as well, though this should be added to the operator package instead.
```

Comment:
does the maturity level of an operator is just a metadata field? 
I would really like to see some defined approach here with testing which can prove that an operator can be upgraded for example. 

What is making KUDO Kafka `1.3.1` as a better operator vs version `0.1.0`?
If we have defined criteria which we agree upon, we could achieve non-biased maturity levels